iot_verifier/src/gateway_cache.rs

```diff
@@ -14,42 +14,53 @@ const CACHE_TTL: Duration = Duration::from_secs(12 * (60 * 60));
 const CACHE_EVICTION_FREQUENCY: Duration = Duration::from_secs(60 * 60);
 
 pub struct GatewayCache {
-    pub follower_service: FollowerService,
+    pub iot_config_client: IotConfigClient,
     pub cache: Arc<Cache<PublicKeyBinary, GatewayInfo>>,
     pub cache_monitor: JoinHandle<()>,
 }
 
-#[derive(thiserror::Error, Debug)]
-#[error("gateway not found: {0}")]
-pub struct GatewayNotFound(PublicKeyBinary);
+#[derive(Debug, thiserror::Error)]
+pub enum GatewayCacheError {
+    #[error("gateway not found: {0}")]
+    GatewayNotFound(PublicKeyBinary),
+    #[error("error querying iot config service")]
+    IotConfigClient(#[from] IotConfigClientError),
+}
 
 impl GatewayCache {
-    pub fn from_settings(settings: &Settings) -> Self {
-        let follower_service = FollowerService::from_settings(&settings.follower);
+    pub fn from_settings(settings: &Settings) -> Result<Self, GatewayCacheError> {
+        let iot_config_client = IotConfigClient::from_settings(&settings.iot_config_client)?;
```

Comment:
i wonder if this is going to cause any problems with the GatewayCache having an IotConfigClient since you're passing the gateway cache into the DensityScaler which _also_ has its own IotConfigClient being init'd from settings

Reply:
perhaps you can init the config client in the main run function and then clone it and pass it to both processes?

Reply:
not sure how tonic handles connections under the hood; since http2 is multiplexed i think you could wrap the underlying h2 connection in an Arc and share it between clients in which case cloning that is just cloning a smart pointer. if it's just a deep clone of the full connection though you're right that it wouldn't be an issue

Reply:
i understand your point now, you are questioning whether we end up with two connections to the same server in the way i have it.   I would seriously doubt that is the case but cannot say for sure.  I will look into.

Tonic does recommend that connections are cloned in order to achieve concurrency, that will be via grpc streams which are multiplexed over the same connection.

Reply:
addressed here: 9f1d2fbd2a617fea9c1d3429ddb6533caf848800

Reply:
I think you can still use same tonic Channel in the gateway cache, the region cache, and the density_scaler if you create a single instance of the IotConfigClient in main.rs with `IotConfigClient::from_settings(&settings.iot_config_client)` and then clone _that_ and pass it as the starting argument to `GatewayCache::from_settings()`, `RegionCache::from_settings()` and `DensityScaler::from_settings()` (the caches' could actually become `::new()` functions instead since the only thing they're doing with their settings is starting the config client